File: backend/src/routes/interview-routes.ts
import { Router } from 'express'
import {
  getInterviewStreamCheckpointController,
  getInterviewSessionDetailController,
  listInterviewSessionsController,
  streamInterviewController
} from '../controllers/interview-controller.js'
import {
  clearInterviewHistoryController
} from '../controllers/interview-history-controller.js'
import {
  generateInterviewReportController,
  getInterviewReportBySessionIdController,
  listInterviewReportsController
} from '../controllers/report-controller.js'
import { generatePracticePoolController } from '../controllers/practice-pool-controller.js'
import { attachOptionalAuth } from '../middlewares/auth-middleware.js'

export const interviewRouter = Router()

interviewRouter.post('/history/clear', clearInterviewHistoryController)
interviewRouter.get('/sessions', attachOptionalAuth, listInterviewSessionsController)
interviewRouter.get('/sessions/:sessionId/:threadId', attachOptionalAuth, getInterviewSessionDetailController)
interviewRouter.get('/sessions/:sessionId/:threadId/checkpoint', attachOptionalAuth, getInterviewStreamCheckpointController)
interviewRouter.post('/stream', attachOptionalAuth, streamInterviewController)
interviewRouter.get('/reports', attachOptionalAuth, listInterviewReportsController)
interviewRouter.get('/reports/:sessionId', attachOptionalAuth, getInterviewReportBySessionIdController)
interviewRouter.post('/reports/generate', attachOptionalAuth, generateInterviewReportController)
interviewRouter.post('/practice-pool/generate', generatePracticePoolController)

File: backend/cpp/src/Routes/InterviewRoutes.cpp
#include "Routes/InterviewRoutes.hpp"

InterviewRoutes::InterviewRoutes(HttpServer& httpServer, InterviewController& controller)
    : m_httpServer(httpServer), m_controller(controller) {}

void InterviewRoutes::registerRoutes() {
    m_httpServer.post("/api/interview/stream",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.streamInterview(req, res);
        });

    m_httpServer.post("/api/interview/reports/generate",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.generateReport(req, res);
        });

    m_httpServer.get("/api/interview/reports",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.listReports(req, res);
        });

    m_httpServer.get("/api/interview/reports/:sessionId",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.getReport(req, res);
        });

    m_httpServer.get("/api/interview/stream",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.streamInterview(req, res);
        });
    m_httpServer.get("/api/interview/sessions",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.listSessions(req, res);
        });

    m_httpServer.get("/api/interview/sessions/:sessionId/:threadId",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.getSession(req, res);
        });

    m_httpServer.post("/api/interview/history/clear",
        [this](const httplib::Request& req, httplib::Response& res) {
            m_controller.clearHistory(req, res);
        });
}
